extension/src/cli/dvc/discovery.ts

```diff
@@ -12,90 +8,82 @@ import {
   getConfigValue,
   setUserConfigValue
 } from '../../vscode/config'
-import { getPythonBinPath } from '../../extensions/python'
 import { getFirstWorkspaceFolder } from '../../vscode/workspaceFolders'
 import { delay } from '../../util/time'
 import { SetupSection } from '../../setup/webview/contract'
 
-export const warnUnableToVerifyVersion = () =>
-  Toast.warnWithOptions(
+const warnWithSetupAction = async (
+  setup: IExtensionSetup,
+  warningText: string
+): Promise<void> => {
+  const response = await Toast.warnWithOptions(warningText, Response.SHOW_SETUP)
+
+  if (response === Response.SHOW_SETUP) {
+    return setup.showSetup(SetupSection.DVC)
+  }
+}
+
+const warnUnableToVerifyVersion = (setup: IExtensionSetup) => {
+  void warnWithSetupAction(
+    setup,
     'The extension cannot initialize as we were unable to verify the DVC CLI version.'
   )
+}
 
-export const warnVersionIncompatible = (
-  version: string,
+const warnVersionIncompatible = (
+  setup: IExtensionSetup,
   update: 'CLI' | 'extension'
 ): void => {
-  void Toast.warnWithOptions(
-    `The extension cannot initialize because you are using version ${version} of the DVC CLI. The expected version is ${MIN_CLI_VERSION} <= DVC < ${MAX_CLI_VERSION}. Please upgrade to the most recent version of the ${update} and reload this window.`
+  void warnWithSetupAction(
+    setup,
+    `The extension cannot initialize because you are using the wrong version of the ${update}.`
   )
 }
 
-export const warnAheadOfLatestTested = (): void => {
+const warnAheadOfLatestTested = (): void => {
   void Toast.warnWithOptions(
     `The located DVC CLI is at least a minor version ahead of the latest version the extension was tested with (${LATEST_TESTED_CLI_VERSION}). This could lead to unexpected behaviour. Please upgrade to the most recent version of the extension and reload this window.`
```

Comment:
Left this toast alone for now since we don't have any info on the latest tested cli version in Setup yet. Once we update our DVC Setup page (task is in #3434 list), we can simplify this one or remove entirely.

Reply:
let's aim to remove it